more/pixels/pixelMintController.js

```diff
@@ -44,66 +63,24 @@ export function makeMintController(assay) {
     // During side effects below, any early exits should be made into
     // fatal turn aborts.
 
-    // we don't know if this is a purse or payment, so handle both cases
-    if (purses.has(asset)) {
-      const originalAmount = purses.get(asset);
-      const newAmount = assay.without(originalAmount, amount);
-      purses.set(asset, newAmount);
-      // Reset the mappings from everything in the amount to the purse
-      // or payment that holds them.
-      recordPixelsAsAsset(newAmount, asset);
-    }
+    const type = asset.getType();
```

Comment:
This seems to be the only place where the code doesn't already know the type of the asset. It's the only place that asks for the type (which is another smell). 

Since the controllers are internal implementation details of the pmController, I think it would be clean enough for it to ask both controllers for the amount, and call updateAmount on whichever responded. I'm sure there are other ways to do it; if you find another approach that you like better (without asking the asset for its type), that would be fine, too.

Reply:
I removed the `getType` from purses and payments and added a `has` method to assetController instead that allows us easily define predicates `isPurse()` and `isPayment()` that can be used in the case of `destroy(amount)`. Does this solve the problem better?

Reply:
As we discussed, this is a big improvement.

If we're going to expose something to help users find out the type of an asset, I prefer the getType() that returns a string rather than a boolean, since that's better for people and harder for code to use.

Reply:
Ok, there is now a getController(asset) method for pixelMintController